### PR TITLE
Topic/lkt coding style

### PR DESCRIPTION
**Pull request by raph-amiard**

### CODE REVIEW

File: doc/lkt_coding_style.rst [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+=====================
+
+No space expected before parens, brackets and colons, for function
+declarations, calls, generic declarations, array accesses, variable
+declarations etc.
+
+.. code-block::
+
+    fun a(): Int = 12
+    generic[T] fun b(): T
```
Comment (pmderodat) [inline doc/lkt_coding_style.rst:41]: This is breaking the rule for generics below

File: doc/lkt_coding_style.rst [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+  parts in the name.
+* Variable names should be lowercase, using underscores when there are several
+  parts.
+
+Parens/brackets/colon
+=====================
+
+No space expected before parens, brackets and colons, for function
+declarations, calls, generic declarations, array accesses, variable
+declarations etc.
```
Comment (pmderodat) [inline doc/lkt_coding_style.rst:36]: I’ve noticed one exception that is worth mentionning: opening paren for function types:

```
foo: (A) -> B
```

File: doc/lkt_coding_style.rst [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+Some flexibility in terms of formatting is granted for annotations. When the
+declaration is short, as in:
+
+.. code-block::
+
+    @parse_field a: B
+
+The annotation can be put on the same line.
+
+When its needed for line lenght reasons to break the code, or when it's felt
```
Comment (pmderodat) [inline doc/lkt_coding_style.rst:75]: it’s*
length*

**Review by pmderodat — CHANGES_REQUESTED** [review]

File: doc/lkt_coding_style.rst (hunk 2 above)
Comment (raph-amiard) [reply doc/lkt_coding_style.rst:36]: Reworded

File: doc/lkt_coding_style.rst [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+* Variable names should be lowercase, using underscores when there are several
+  parts.
+
+Parens/brackets/colon
+=====================
+
+* No space expected before parens for calls and arrays accesses
+* No space expected before brackets for generic declarations and
+  instantiations.
+* No Space expected before colons for type annotations, but there should always
```
Comment (pmderodat) [inline doc/lkt_coding_style.rst:37]: space*

**Review by pmderodat — APPROVED** [review]
Minor typo left, then it’s good to go, thanks!

**Review by pmderodat — APPROVED** [review]

**Pull request merged**